src/ol/reproj/image.js

```diff
@@ -0,0 +1,203 @@
+goog.provide('ol.reproj.Image');
+
+goog.require('goog.asserts');
+goog.require('goog.events');
+goog.require('goog.events.EventType');
+goog.require('ol.ImageBase');
+goog.require('ol.ImageState');
+goog.require('ol.extent');
+goog.require('ol.proj');
+goog.require('ol.reproj');
+goog.require('ol.reproj.Triangulation');
+
+
+
+/**
+ * @classdesc
+ * Class encapsulating single reprojected image.
+ * See {@link ol.source.Image}.
+ *
+ * @constructor
+ * @extends {ol.ImageBase}
+ * @param {ol.proj.Projection} sourceProj Source projection (of the data).
+ * @param {ol.proj.Projection} targetProj Target projection.
+ * @param {ol.Extent} targetExtent Target extent.
+ * @param {number} targetResolution Target resolution.
+ * @param {number} pixelRatio Pixel ratio.
+ * @param {function(ol.Extent, number, number):ol.ImageBase} getImageFunction
+ *     Function returning source images (extent, resolution, pixelRatio).
```

Comment:
You could also create a `@typedef` to document the signature of the function, for example like here:
https://github.com/openlayers/ol3/blob/master/src/ol/centerconstraint.js#L7-L10